### PR TITLE
IGNITE-14923

### PR DESCRIPTION
**Pull request by tkalkirill**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalArchiveSizeConfigurationTest.java [hunk 1]
```diff
@@ -103,6 +113,39 @@ public void testUnlimitedMaxArchiveSizeConfiguration() throws Exception {
...
+    public void testIncorrectMinArchiveSizeConfiguration() {
+        DataStorageConfiguration dsCfg = new DataStorageConfiguration()
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration().setPersistenceEnabled(true))
+            .setMinWalArchiveSize(Long.MAX_VALUE);
+
+        assertThrowsAnyCause(
+            log,
+            () -> startGrid(0, (IgniteConfiguration cfg) -> cfg.setDataStorageConfiguration(dsCfg)),
+            IgniteCheckedException.class,
+            "DataRegionConfiguration.minWalArchiveSize must be less than or equal to"
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalArchiveSizeConfigurationTest.java:129]: what?

File: modules/core/src/main/java/org/apache/ignite/configuration/DataStorageConfiguration.java [hunk 2]
```diff
@@ -188,6 +188,9 @@
     /** @see IgniteSystemProperties#IGNITE_USE_ASYNC_FILE_IO_FACTORY */
     public static final boolean DFLT_USE_ASYNC_FILE_IO_FACTORY = true;
 
+    /** Value used to indicate the use of half of the {@link #getMaxWalArchiveSize}. */
+    public static final long HALF_MAX_WAL_ARCHIVE_SIZE = -2;
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/configuration/DataStorageConfiguration.java:192]: Please replace it with -1

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalArchiveSizeConfigurationTest.java (hunk 1 above)
Comment (ibessonov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalArchiveSizeConfigurationTest.java:129]: sorry, everything's fine here

Comment (tkalkirill) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalArchiveSizeConfigurationTest.java:129]: This is only for finding a substring of the error text

**Pull request merged**